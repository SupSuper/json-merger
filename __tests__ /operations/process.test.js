const jsonMerger = require("../../dist");
const {testConfig} = require("../__helpers__");

describe("when merging two objects and a source property has a $process indicator", function () {

    test("it should process the $process property and then process/merge the result with the target", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$process": {
                    "$$replace": "replaced"
                }
            }
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("it should use the $process property result as $currentSource when processing/merging with the target", function () {

        const object = {
            "$process": {
                "$merge": {
                    "source": {
                        "a": {
                            "$$expression": "$currentSource.b"
                        }
                    },
                    "with": {
                        "b": "bValue"
                    }
                }
            }
        };

        const result = jsonMerger.mergeObject(object, testConfig());

        expect(result).toMatchSnapshot();
    });
});
