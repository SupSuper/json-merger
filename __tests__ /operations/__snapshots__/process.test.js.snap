// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two objects and a source property has a $process indicator it should process the $process property and then process/merge the result with the target 1`] = `
"{
	\\"a\\": \\"replaced\\"
}"
`;

exports[`when merging two objects and a source property has a $process indicator it should use the $process property result as $source when processing/merging with the target 1`] = `
"{
	\\"a\\": \\"bValue\\",
	\\"b\\": \\"bValue\\"
}"
`;
